src/js/opponent.js

```diff
@@ -22,75 +22,214 @@ export default class Opponent extends Player {
     constructor(cards = []) {
       super(cards);
     }
-
-    findDoubles() {
-      var doubles = [];
     
-      for (let i = 0; i < this.numberOfCards - 1; i++) {
+    cardCmp(a,b){
+      //cmpFunc for combos prioritizing length then values ORDER:Desc
+      var lengthDelta = a.length - b.length;
+      if (lengthDelta == 0){
+        return b[0].value - a[0].value;
+
+      }else{
+        return lengthDelta;
+      }
+    }
+
+    findDupes(){
+      //find n-card multiples; save time on combo searching
+      this.cards.sort((a,b) => a.value - b.value);
+      var dupes = [];
+      var curVal = NaN
+      for (let i = 0; i < this.numberOfCards; i++) {
+        var currentCombo = [];
         var currentCard = this.cards[i];
-        var nextCard = this.cards[i + 1];
-        
-        if (currentCard.value === nextCard.value) {
-          // check if cards have the same value but different suits
-          if (currentCard.suit !== nextCard.suit) {
-            doubles.push(currentCard);
-            doubles.push(nextCard);
-            i++; // skip the next card since it has already been considered as a double
-          }
+        curVal = currentCard.value;
+        currentCombo.push(currentCard);
+        var nextCardIndex = i+1;
+        var nextCard = this.cards[nextCardIndex];
+
+        while (nextCard.value == curVal){
```

Comment:
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'value')
    at Opponent.findDupes (opponent.js:50:25)
    at Opponent.selectCard (opponent.js:252:24)
    at Opponent.playCard (opponent.js:337:25)
    at forLoop (big2.js:126:46)
    at async startGame (big2.js:164:22)